flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/catalog/AbstractJdbcCatalog.java

```diff
@@ -522,6 +509,25 @@ protected List<String> extractColumnValuesBySQL(
         }
     }
 
+    protected static List<String> extractColumnValuesByStatement(
+            PreparedStatement ps, int columnIndex, Predicate<String> filterFunc, Object... params)
+            throws SQLException {
+        List<String> columnValues = Lists.newArrayList();
+        if (Objects.nonNull(params) && params.length > 0) {
+            for (int i = 0; i < params.length; i++) {
+                ps.setObject(i + 1, params[i]);
+            }
+        }
+        ResultSet rs = ps.executeQuery();
```

Comment:
Nit Pick: though not related to your change but can you wrap the above line in try with resource block I think ResultSet also implements AutoClosable

Reply:
`try(ResultSet rs = ps.executeQuery()) {...}`

Reply:
Sounds good. Updated. thanks